src/lib/FreeC/LiftedIR/Converter/Expr.hs

```diff
@@ -326,3 +325,25 @@ rawBind srcSpan mx x varType expr = do
       Just unqualX = identFromQName x
       x'           = LIR.VarPat srcSpan unqualX varType' xAgda xCoq
   return $ LIR.Bind srcSpan mx' $ LIR.Lambda srcSpan [x'] expr
+
+-------------------------------------------------------------------------------
+-- Let Expression Helper                                                     --
+-------------------------------------------------------------------------------
+-- | Lifts a list of let bindings.
+--
+--   The given expression is the right-hand side of the let.
+liftBinds :: [IR.Bind] -> IR.Expr -> Converter LIR.Expr
+liftBinds [] expr = liftExpr expr
+liftBinds
+  ((IR.Bind srcSpan (IR.VarPat patSrcSpan ident varPatType isStrict) bindExpr)
+   : bs) expr = do
```

Comment:
To avoid the variable to be visible outside of the `let` expression, a local environment must be used here.

```suggestion
   : bs) expr = localEnv $ do
```